File: TinyScript/js/src/main/js/parser/ast/Variable.ts
import Factor from "./Factor";
import ASTNodeTypes from "./ASTNodeTypes";
import Token from "../../lexer/Token";

export default class Variable extends Factor {
        private _typeLexeme:Token | null;


    constructor(token: Token) {
        super(token);
        this.type = ASTNodeTypes.VARIABLE;
        this._typeLexeme = null;
    }


    get typeLexeme(): Token | null {
        return this._typeLexeme;
    }

    set typeLexeme(value: Token | null) {
        this._typeLexeme = value;
    }
}

File: TinyScript/java/src/main/java/paser/ast/Variable.java
package paser.ast;

import lexer.Token;
import paser.util.PeekTokenIterator;

/**
 * @author MFine
 * @version 1.0
 * @date 2021/8/23 0:20
 **/
public class Variable extends Factor {

    private Token typeLexeme = null;

    public Variable(ASTNode parent, PeekTokenIterator it) {
        super(parent, it);
    }

    public Token getTypeLexeme() {
        return typeLexeme;
    }

    public void setTypeLexeme(Token typeLexeme) {
        this.typeLexeme = typeLexeme;
    }
}
